sdk/core/Azure.Core/src/ClientOptions.cs

```diff
@@ -132,5 +150,16 @@ public void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position)
         /// <inheritdoc />
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override string? ToString() => base.ToString();
+
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        public override void Freeze()
+        {
+            Diagnostics.Freeze();
```

Comment:
Interesting!  I hadn't thought of that, but it's a good call.  Let me investigate.  We may also decide to break the inheritance relationship between `ClientOptions` and `ClientPipelineOptions` -- if we do, this would become a non-issue because we just wouldn't make `ClientOptions` freezable.

Reply:
Removed `ClientOptions` inheritance from this PR.